src/lib/mina_ledger/lmdb_kvdb.ml

```diff
@@ -0,0 +1,95 @@
+open Core_kernel
+
+module Kvdb = struct
+  type t =
+    { mutable lmdb :
+        ((Bigstring.t, Bigstring.t, [ `Uni ]) Lmdb.Map.t[@sexp.opaque])
+    ; path : string
+    ; uuid : (Uuid.t[@sexp.opaque])
+    }
+  [@@deriving sexp]
+
+  let trivial_conv : Bigstring.t Lmdb.Conv.t =
+    Lmdb.Conv.make ~serialise:(Fn.const Fn.id) ~deserialise:Fn.id ()
+
+  (* TODO is this good enough? what other data can I use to make it more trustably unique? *)
+  let name_to_uuid name =
+    String.to_array name |> Array.map ~f:Char.to_int |> Random.State.make
+    |> Uuid.create_random
```

Comment:
For the sake of nitpicking (avoiding one array allocation) and documentation (another implementation)
```suggestion
    Array.init (String.length name) (fun i -> Char.to_int (String.get name i))
    |> Random.State.make
    |> Uuid.create_random
```